.github/workflows/smoke-test.yaml

```diff
@@ -0,0 +1,182 @@
+# This workflow downloads artifacts from a (by default, draft) release and runs
+# a short smoke test where the application is installed and run and immediately
+# shut down.
+# Since we need contents-write permissions to look at draft releases, we
+# actually download the artifacts in a smaller job, then upload them into the
+# run and download it _again_ in the second (per-platform) job where no
+# permissions are required.
+name: Release smoke test
+permissions: {}
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: >
+          Download artifacts from release with this tag, rather than picking the
+          first draft release.
+        type: string
+
+jobs:
+  download-artifacts:
+    name: Find release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # Needed to list draft releases
+    env:
+      RELEASE_TAG: ${{ inputs.tag }}
+    steps:
+    - name: Find release
+      if : inputs.tag == ''
+      run: >-
+        set -o xtrace;
+        printf "RELEASE_TAG=%s\n" >>"$GITHUB_ENV"
+        "$(gh api repos/${{ github.repository }}/releases
+        --jq 'map(select(.draft))[0].tag_name')"
+      env:
+        GH_TOKEN: ${{ github.token }}
+    - name: Download artifacts
+      run: |
+        if [[ -z "$RELEASE_TAG" ]]; then
+          echo "Failed to find release tag" >&2
+          exit 1
+        fi
+        # For macOS, currently only x86_64 supports nested virtualization
```

Comment:
Instead of skipping macOS on aarch64 completely, we could still do the download and unpack tests.

Maybe there is also a way to run the app in the "blocked" state (pretending there is a snapshot operation in progress), so the GUI will start, but not run the Lima VM. Not a full test, but better than nothing.

Reply:
> Maybe there is also a way to run the app in the "blocked" state (pretending there is a snapshot operation in progress), so the GUI will start, but not run the Lima VM. Not a full test, but better than nothing.

On second thoughts this doesn't sound too useful. Just unpacking the bits and checking the signatures is enough.